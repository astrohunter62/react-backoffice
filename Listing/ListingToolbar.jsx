import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'

import { withStyles } from 'material-ui/styles'
import Toolbar from 'material-ui/Toolbar'
import Tooltip from 'material-ui/Tooltip'
import Typography from 'material-ui/Typography'
import IconButton from 'material-ui/IconButton'

import DeleteIcon from 'material-ui-icons/Delete'
import ListingSearch from './ListingSearch'

const toolbarStyles = theme => ({
  root: {
    paddingRight: 2,
  },
  highlight:
    theme.palette.type === 'light'
      ? {
        color: theme.palette.secondary.dark,
        backgroundColor: theme.palette.secondary.light,
      }
      : {
        color: theme.palette.secondary.light,
        backgroundColor: theme.palette.secondary.dark,
      },
  spacer: {
    flex: '1 1 100%',
  },
  actions: {
    color: theme.palette.text.secondary,
  },
  title: {
    flex: '0 0 auto',
  },
})

const ListingToolbar = ({
  title,
  numSelected,
  onFilter,
  classes,
}) => (
  <Toolbar
    className={classNames(classes.root, {
        [classes.highlight]: numSelected > 0,
      })}
  >
    <div className={classes.title}>
      {numSelected > 0 ? (
        <Typography type="subheading">{numSelected} selected</Typography>
      ) : (
        <Typography type="title">{title}</Typography>
      )}
    </div>
    <div className={classes.spacer} />
    <div className={classes.actions}>

      {numSelected > 0 ? (
        <Tooltip title="Delete">
          <IconButton aria-label="Delete">
            <DeleteIcon />
          </IconButton>
        </Tooltip>
      ) : (
        <ListingSearch onFilter={onFilter} />
      )}
    </div>
  </Toolbar>
)

ListingToolbar.propTypes = {
  title: PropTypes.string.isRequired,
  numSelected: PropTypes.number.isRequired,
  onFilter: PropTypes.func.isRequired,
  classes: PropTypes.objectOf(PropTypes.string).isRequired,
}

export default withStyles(toolbarStyles)(ListingToolbar)
