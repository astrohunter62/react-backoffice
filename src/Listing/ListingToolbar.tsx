import React, { FunctionComponent } from "react";
import classNames from "classnames";
import { makeStyles, Theme, Toolbar, Typography } from "@material-ui/core";
import ListingSearch from "./ListingSearch";

const useStyles = makeStyles((theme: Theme) => ({
  root: {
    paddingRight: 2,
  },
  highlight:
    theme.palette.type === "light"
      ? {
          color: theme.palette.secondary.dark,
          backgroundColor: theme.palette.secondary.light,
        }
      : {
          color: theme.palette.secondary.light,
          backgroundColor: theme.palette.secondary.dark,
        },
  spacer: {
    flex: "1 1 100%",
  },
  actions: {
    color: theme.palette.text.secondary,
  },
  title: {
    flex: "0 0 auto",
  },
}));

type ListingToolbarProps = {
  title?: string;
  numSelected?: number;
  onFilter?: (...args: any[]) => any;
};

const ListingToolbar: FunctionComponent<ListingToolbarProps> = ({
  title,
  numSelected = 0,
  onFilter,
  children,
}) => {
  const classes = useStyles();

  return (
    <Toolbar
      className={classNames(classes.root, {
        [classes.highlight]: numSelected > 0,
      })}
    >
      <div className={classes.title}>
        {numSelected > 0 ? (
          <Typography variant="subtitle1">
            {numSelected}
            &nbsp;selected
          </Typography>
        ) : (
          <Typography variant="h6">{title}</Typography>
        )}
      </div>

      <div className={classes.spacer} />

      <div className={classes.actions}>
        {numSelected > 0 ? children : <ListingSearch onFilter={onFilter} />}
      </div>
    </Toolbar>
  );
};

export default ListingToolbar;
