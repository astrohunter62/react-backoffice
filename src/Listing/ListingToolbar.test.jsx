import React from 'react'
import renderer from 'react-test-renderer'

import ListingToolbar from './ListingToolbar'

it('renders correctly', () => {
  const tree = renderer
    .create((
      <ListingToolbar
        title=""
        numSelected={0}
        onFilter={() => { }}
      />
    ))
    .toJSON()
  expect(tree).toMatchSnapshot()
})
