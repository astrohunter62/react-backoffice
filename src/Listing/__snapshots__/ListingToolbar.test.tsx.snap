// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ListingToolbar
  classes={
    Object {
      "actions": "ListingToolbar-actions-4",
      "highlight": "ListingToolbar-highlight-2",
      "root": "ListingToolbar-root-1",
      "spacer": "ListingToolbar-spacer-3",
      "title": "ListingToolbar-title-5",
    }
  }
  numSelected={0}
  onFilter={[Function]}
  title=""
/>
`;
